internal/metrics/system.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"github.com/jakub-dzon/k4e-device-worker/internal/configuration"
+	"github.com/jakub-dzon/k4e-operator/models"
+	"reflect"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	DefaultSystemMetricsScrapingInterval = 60
+	NodeExporterMetricsEndpoint          = "http://localhost:9100/metrics"
+)
+
+var defaultSystemMetricsConfiguration = models.SystemMetricsConfiguration{Interval: DefaultSystemMetricsScrapingInterval}
+
+//go:generate mockgen -package=metrics -destination=configuration_provider_mock.go . DeviceConfigurationProvider
+type DeviceConfigurationProvider interface {
+	GetDeviceConfiguration() models.DeviceConfiguration
+}
+
+type SystemMetrics struct {
+	latestConfig atomic.Value
+	daemon       MetricsDaemon
+	config       configuration.Manager
+}
+
+func NewSystemMetrics(daemon MetricsDaemon, config DeviceConfigurationProvider) *SystemMetrics {
+	expectedConfiguration := expectedConfiguration(config.GetDeviceConfiguration())
+
+	sm := &SystemMetrics{
+		daemon: daemon,
+	}
+	daemon.AddTarget("system", []string{NodeExporterMetricsEndpoint}, time.Duration(expectedConfiguration.Interval)*time.Second, nil)
```

Comment:
I would move the "system" to a const, also added a if condition on metrics/workload if any the name is the same to log an error.

Reply:
const - done, validation will be implemented separately in k4e-operator